cairo-m/runner/src/lib.rs

```diff
@@ -1,64 +1,127 @@
 uniffi::setup_scaffolding!();
 
 use cairo_m_common::Program;
+use cairo_m_prover::{
+    adapter::import_from_runner_output, prover::prove_cairo_m, verifier::verify_cairo_m,
+};
 use cairo_m_runner::run_cairo_program;
+// use cairo_m_prover::verifier::verify_cairo_m;
+use stwo_prover::core::vcs::blake2_merkle::Blake2sMerkleChannel;
 
 /// Represents the possible errors that can occur in the mobile VM.
 #[derive(Debug, thiserror::Error, uniffi::Error)]
-pub enum MobileVmError {
-    #[error("VM Error: {0}")]
-    Vm(String),
+pub enum MobileError {
     #[error("JSON parsing error: {0}")]
     Json(String),
+    #[error("VM Error: {0}")]
+    Vm(String),
+    #[error("Proof generation error: {0}")]
+    Proof(String),
+    #[error("Verification error: {0}")]
+    Verification(String),
 }
 
-/// The result of a successful program execution.
+/// The result and metrics of a successful program execution and proof generation.
 ///
 /// # Fields
 ///
 /// * `return_value` - The return value of the program
-/// * `frequency` - The frequency of the trace generation, in Hz
+/// * `overall_frequency` - The frequency of the execution and proof generation, in Hz
+/// * `execution_frequency` - The frequency of the execution, in Hz
+/// * `proof_frequency` - The frequency of the proof generation, in Hz
+/// * `proof_size` - The size of the proof, in bytes
+/// * `proof` - The proof of the program, serialized as a JSON string
 #[derive(Debug, uniffi::Record)]
-pub struct RunResult {
+pub struct RunProofResult {
     pub return_value: u32,
-    pub frequency: f64,
+    pub overall_frequency: f64,
+    pub execution_frequency: f64,
+    pub proof_frequency: f64,
+    pub proof_size: u32,
+    pub proof: String,
+}
+
+#[derive(Debug, uniffi::Record)]
+pub struct VerifyResult {
+    pub verification_time: f64,
 }
 
 /// Runs a compiled Cairo program and returns the result and execution metrics.
```

Comment:
Something like that?
```suggestion
/// Runs a compiled Cairo program and generate a proof of execution. It returns the result, execution metrics and the proof generated.
```